### PR TITLE
draft 1st vignette

### LINKED ISSUE #10: `tbl_adverse_events()` define & provide rationale behind counting rules

### PR DESCRIPTION
**Pull request by shannonpileggi**
**What changes are proposed in this pull request?**
- added first vignette for counting methods
- ran `devtools::document()`
- ran `pkgdown::build_site()`

**Discussion**

- not sure it worked entirely correctly? don't see vignette pkgdown preview
- any features to add or take away from example data to make counting rules clearer?
- will take any other feedback of course!

**If there is an GitHub issue associated with this pull request, please provide link.**

- closes #10 

--------------------------------------------------------------------------------

Checklist for PR reviewer

- [ ] Ensure all package dependencies are installed by running `renv::install()`
- [ ] PR branch has pulled the most recent updates from main branch. Ensure the pull request branch and your local version match and both have the latest updates from the main branch.
- [ ] If a new function was added, function included in `_pkgdown.yml`
- [ ] If a bug was fixed, a unit test was added for the bug check
- [ ] Run `pkgdown::build_site()`. Check the R console for errors, and review the rendered website.
- [ ] Code coverage is suitable for any new functions/features. Review coverage with `covr::report()`. Before you run, set `Sys.setenv(NOT_CRAN="true")` and begin in a fresh R session without any packages loaded. 
- [ ] R CMD Check runs without errors, warnings, and notes
- [ ] `usethis::use_spell_check()` runs with no spelling errors in documentation
- [ ] Has `NEWS.md` been updated with the changes from this pull request under the heading "`# gtreg (development version)`". If there is an issue associated with the pull request, reference it in parentheses at the end update (see `NEWS.md` for examples).
- [ ] Has the version number been incremented using `usethis::use_version(which = "dev")` 
- [ ] Approve Pull Request
- [ ] Merge the PR. Please use "Squash and merge".

### CODE REVIEW

File: vignettes/counting-methods.Rmd [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+dat %>% 
+  tbl_ae(
+    id = subject,
+    id_df = tibble::tibble(subject = c("001", "002", "003")),
+    ae = ae,
+    soc = soc,
+    by = grade
+  ) %>% 
+  modify_ae_header(all_ae_cols() ~ "**Grade {by}**") %>% 
+  modify_ae_spanning_header(all_ae_cols() ~ "**N = {N}**") %>%
```
Comment (ddsjoberg) [inline vignettes/counting-methods.Rmd:152]: The N you're adding here applies to the overall column too, but it's only being placed above the by levels. Likely doesn't matter when the default is updated to include the N

File: vignettes/counting-methods.Rmd [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+
+Reporting of adverse events to regulatory agencies typically employs both 
+standardized terminology and specific counting methods for adverse events. 
+
+# Standardized Terminology
+
+Providing standardized terminology is not in the scope of this package. Standardized
+terminology adheres to a hierarchy of terms from specific to general, as discussed
+in the [MedDRA Hierarchy](https://www.meddra.org/how-to-use/basics/hierarchy) 
+documentation. In `gtreg`, we facilitate reporting of a single term
```
Comment (ddsjoberg) [inline vignettes/counting-methods.Rmd:34]: In the README, the package name is wrapped in curly brackets, but here is in backticks. We should try to be consistent throughout

File: vignettes/counting-methods.Rmd [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+  modify_ae_header(all_ae_cols() ~ "**Grade {by}**") %>% 
+  add_overall() %>% 
+  bold_labels()
+```
+
+In this case, we have two subjects which experienced a total of 5 eye disorders and 4
+gastrointestinal disorders over the course of multiple time points. The only statistic 
+allowed in `tbl_ae_count()` is `n`.
+
+# Counting By Maximum
```
Comment (ddsjoberg) [inline vignettes/counting-methods.Rmd:84]: Maybe "Counting By Maximum Grade" ?

**Review by ddsjoberg — CHANGES_REQUESTED** [review]
Great vignette! Makes the methods very clear.

Do you want to add a `#' @seealso` line to `tbl_ae()` and `tbl_ae_count()` linking to this vignette?

> not sure it worked entirely correctly? don't see vignette pkgdown preview

The `pkgdown_yaml` needs to be updated to link to the new vignette. Under the `navbar:` section add this:
```
  - text: Articles
    menu:
    - text: "Counting Rules"
      href: articles/counting-methods.html
```

File: vignettes/counting-methods.Rmd (hunk 2 above)
Comment (shannonpileggi) [reply vignettes/counting-methods.Rmd:34]: Good point! I have changed to curly brackets.

File: vignettes/counting-methods.Rmd (hunk 3 above)
Comment (shannonpileggi) [reply vignettes/counting-methods.Rmd:84]: Sounds good. I also added another paragraph to address `by` as factor.

**Comment by shannonpileggi** [comment]
Thanks for the pkgdown tips! We might need a separate issue later to review pkgdown site more thoroughly.

**Comment by shannonpileggi** [comment]
I think we can hold off on this until we get a couple of more PRs merged.

**Comment by shannonpileggi** [comment]
- added in grade unknown events in data table and mentioned in counting methods.
- changed `Counting By Maximum Grade`  to `Counting By Highest Grade` 
- added details about counting system organ class

**Comment by shannonpileggi** [comment]
@ddsjoberg thanks for the addition! can you approve and then i'll take a look at docs again w/ fresh eyes?

**Review by ddsjoberg — APPROVED** [review]
<3

**Pull request merged**